react/__tests__/components/BuyButton.test.js

```diff
@@ -34,4 +43,32 @@ describe('<BuyButton />', () => {
     const { asFragment } = renderComponent({ available: false })
     expect(asFragment()).toMatchSnapshot()
   })
+
+  it('should call onAddStart', () => {
```

Comment:
I think both test cases can be merged into one that will garantee that `onAddStart` is called then `onAddEnd` is called.